src/components/dashboard/DashboardDetail.js

```diff
@@ -133,60 +76,62 @@ class DashboardDetail extends Component {
                </Link>
             )}
          >
-            <DashboardKeys
-               image={AuthorizedGreyImage}
-               title='Authorized Apps'
-               to='/authorized-apps'
-               accessKeys={authorizedApps}
-            />
-            <DashboardKeys
-               image={AuthorizedGreyImage}
-               title='Full Access Keys'
-               to='/full-access-keys'
-               accessKeys={fullAccessKeys}
-            />
-            {false ? (
-               <DashboardSection
-                  notice={notice}
-                  handleNotice={this.handleNotice}
-               >
-                  <DashboardActivity
-                     loader={loader}
-                     image={activityGreyImage}
-                     title='Activity'
-                     to='/'
-                     activity={activity}
-                  />
-                  <DashboardActivity
-                     loader={loader}
-                     image={AuthorizedGreyImage}
-                     title='Authorized Apps'
-                     to='/authorized-apps'
-                     activity={authorizedApps}
-                  />
-                  <DashboardActivity
-                     loader={loader}
-                     image={ContactsGreyImage}
-                     title='New Contacts'
-                     to='/contacts'
-                     activity={newcontacts}
-                  />
-               </DashboardSection>
-            ) : null}
+            <DashboardSection
+               notice={notice}
+               handleNotice={this.handleNotice}
+            >
+               <DashboardActivity
+                  loader={loader}
+                  image={activityGreyImage}
+                  title='Activity'
+                  to={`${process.env.EXPLORER_URL || 'https://explorer.nearprotocol.com'}/accounts/${accountId}`}
+                  transactions={transactions}
+               />
+               <DashboardKeys
+                  image={AuthorizedGreyImage}
+                  title='Authorized Apps'
+                  to='/authorized-apps'
+                  accessKeys={authorizedApps}
+               />
+               <DashboardKeys
+                  image={AuthorizedGreyImage}
+                  title='Full Access Keys'
+                  to='/full-access-keys'
+                  accessKeys={fullAccessKeys}
+               />
+            </DashboardSection>
          </PageContainer>
       )
    }
 }
 
 const mapDispatchToProps = {
-   getAccessKeys
+   getAccessKeys,
+   getTransactions
 }
 
-const mapStateToProps = ({ account }) => ({
-   ...account,
-   authorizedApps: account.authorizedApps,
-   fullAccessKeys: account.fullAccessKeys
-})
+const mapStateToProps = ({ account }) => {
+   const transactions = account.transactions 
+      ? account.transactions.reduce((x, t) => {
+            try {
+               t.actions = JSON.parse(t.actions)
```

Comment:
feels like in pure functional style you want something like:
```
t = { ...t, actions: JSON.parse(t.actions) }
```

Reply:
@vgrichina I know JS devs don't care, but you shallow copy `t` that way, which is not the case in the original implementation.

Reply:
@frol 
> but you shallow copy t that way

That is the whole point, that what should happen inside functional-style code unless you have good performance-optimization reasons.

BTW, there are even some cases where creating copies results in faster code. E.g. if you use immutable data structures for everything (not the case here), it'll take less time to determine what has changed between re-renders.

Reply:
To be honest, I avoid reduce/fold operators in non-functional languages as a simple for loop is usually much readable.

Reply:
> To be honest, I avoid reduce/fold operators in non-functional languages as a simple for loop is usually much readable.

JS is a functional language though (just very unsure), with object system more or less built on top :)

@frol @marcinbodnar however a good point is that `.reduce` usually not the most clear way to express stuff, like my understanding is that here you actually just want to `.map` transactions list first and then join arrays e.g. using `[].concat.apply([], listOfArrays)`. `.reduce` is a bit an overkill to join arrays.

Reply:
@vgrichina ok, I can use `.map` and some additional variable to store new transactions array. 

btw I think `.reduce` is very handy for lops when you need one additional variable.

Reply:
flatMap seems to be more appropriate here

Reply:
@frol didn't even realize `.flatMap` is already a thing in modern browsers :)